src/kOS.Safe/Compilation/KS/Compiler.cs

```diff
@@ -341,7 +361,7 @@ private void PreProcessWhenStatement(ParseNode node)
         /// </summary>
         private string ConcatenateNodes(ParseNode node)
         {
-            return string.Format("{0}{1}", context.NumCompilesSoFar, ConcatenateNodesRecurse(node));
+            return string.Format("{0}{1}{2}", context.NumCompilesSoFar, GetContainingScopeId(node), ConcatenateNodesRecurse(node));
```

Comment:
Added the extra term because basically the attempt to mash together different instances of the same expression into the same function or trigger isn't so reliable when using closures - the same expression in these braces might not REALLY be the same as that expression over there in those other braces becasue they need to have a different scope ID for their locals.  So mostly the entire concept of these hashcodes is nearly pointless now.  We could change the hashcode to just something that simply is unique for "this exact location in the source file of this exact compile instance" and stop paying attention to the contents of the node.